amazonorders/orders.py

```diff
@@ -13,8 +13,9 @@
 
 logger = logging.getLogger(__name__)
 
-ORDER_HISTORY_CARD_SELECTOR = "div[class*='order-card']:has(script)"
+ORDER_HISTORY_CARD_SELECTOR = "div[class*='js-order-card']"
 ORDER_DETAILS_DIV_SELECTOR = "div[id='orderDetails']"
+NEXT_PAGE_LINK = "{}/gp/your-account/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-{}{}"
```

Comment:
We don't want to use this URL, because it's not queryable. From your Amazon page, go to Order History, then select the year dropdown and pick 2023. It should take you to a different URL that has a `timeFilter` in the query string.

The default order history page only ever shows the last three months of orders, and parses differently, whereas the year filter page the library is using (at present, anyway) is more consistent. Try using that page and see if/how it's different on your account.

Reply:
> We don't want to use this URL, because it's not queryable. From your Amazon page, go to Order History, then select the year dropdown and pick 2023. It should take you to a different URL that has a `timeFilter` in the query string.
^^ the route you describe is how I got to that link. The link above still uses the `orderFilter=year-`

> The default order history page only ever shows the last three months of orders, and parses differently, whereas the year filter page the library is using (at present, anyway) is more consistent. Try using that page and see if/how it's different on your account.

Reply:
Okay, so I noticed yours has `orderHistory` as the query param, mine has `timeFilter`. Fun fact, if I manually change my URL to `orderFilter`, the page loads and looks the same, but the URL redirect suddenly has the word "legacy" in it:

`https://www.amazon.com/gp/legacy/order-history?orderFilter=year-2024&ref_=ppx_yo2ov_dt_b_filter_all_y2024&`

What I'm assuming are the new URLs have gotten rid of /gp/legacy and time `orderFilter` to `timeFilter`. Visually the pages looked the same, but under the hood they must be different.

Reply:
Woah there with the assumptions… 😉 When I go to the route you provided, I get redirected to:
https://www.amazon.com/gp/legacy/order-history?timeFilter=year-2023&

Reply:
Ha, that's amazing. It must be A/B testing then. _One_ time a week or so back, when I was running an integration test, I got back a page asking me if I wanted to opt in to the new Orders page. I couldn't interact with the prompt since it was in a test, and I've yet to see it in my browser, but it did lead me to believe they're probably in the process of breaking everything I'm doing 🤓.